File: CivitasJava/civitas/CasillaCalle.java
package civitas;

import java.util.ArrayList;

public class CasillaCalle extends Casilla {
    
    // Atributos
    TituloPropiedad tituloPropiedad;
    
    // Métodos
    CasillaCalle (TituloPropiedad titulo){
        super(titulo.getNombre());
        tituloPropiedad = titulo;
    }
            
    TituloPropiedad getTituloPropiedad(){
        return tituloPropiedad;
    }
              
    @Override
    void recibeJugador (int iactual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(iactual, todos)){
            informe(iactual, todos);
            Jugador jugador = todos.get(iactual);
              
            if (!tituloPropiedad.tienePropietario()){
		        jugador.puedeComprarCasilla();
            }else{
                tituloPropiedad.tramitarAlquiler(jugador);
            };
        };
    };
};

File: CivitasRuby/casilla_calle.rb
require_relative 'casilla'

module Civitas

class CasillaCalle < Casilla
  
  def initialize(titulo)
    super(titulo.nombre)
    @tituloPropiedad = titulo
  end
  
  def recibe_jugador(actual, todos)
    if jugador_correcto(actual, todos)
      informe(actual, todos)
      
      if !@tituloPropiedad.tiene_propietario
        todos[actual].puede_comprar_casilla
      else
        @tituloPropiedad.tramitar_alquiler(todos[actual])
      end
    end
  end
  
  attr_reader :tituloPropiedad
end
end
